OSDOCS-9589: Update Architecture Models Sub-Section to Show the Differences between Classic and HCP (#72358)
openshift/openshift-docs: modules/rosa-architecture.adoc

// Module included in the following assemblies:
//
// * rosa_architecture/rosa_architecture_sub/rosa-architecture-models.adoc

[id="rosa-classic-architecture_{context}"]
= ROSA Classic architecture

In ROSA Classic, both the control plane and the worker nodes are deployed on the customer's VPC subnets.

[id="rosa-classic-architecture-networks_{context}"]
== ROSA Classic architecture on public and private networks

With ROSA Classic, you can create your clusters on public or private networks.

> Customers can customize access patterns for their API server endpoint and Red Hat SRE management in the following ways:

Reviewer: In the preview, the bullet points are not wrapped/formatted. can you please check?

// Module included in the following assemblies:
//
// * rosa_architecture/rosa_architecture_sub/rosa-architecture-models.adoc

[id="rosa-classic-architecture_{context}"]
= ROSA Classic architecture

> In ROSA Classic, both the control plane and the worker nodes are deployed on the customer's VPC subnets.

Reviewer 2: ```suggestion
In ROSA Classic, both the control plane and the worker nodes are deployed in the customer's VPC subnets.
```

[id="rosa-classic-architecture-networks_{context}"]
== ROSA Classic architecture on public and private networks

> With ROSA Classic, you can create your clusters on public or private networks.

Reviewer 2: ```suggestion
With ROSA Classic, you can create your clusters that are accessible over public or private networks.
```
Trying to avoid some confusion because ROSA classic clusters can't be created in public subnets - the label of public/private relates to how to access the clusters' API server/ingress

Reviewer: Original statement mentioned networks that could broadly to subnets too. Michael's suggestion is better because clusters are always deployed in private subnets and it's accessibility that's public or private.

Customers can customize access patterns for their API server endpoint and Red Hat SRE management in the following ways:
* Public - API server endpoint and application routes are internet-facing.

> * Private - API server endpoint and application routes are private. Private ROSA Classic clusters use some public subnets, but no control plane or worker nodes are deployed in public subnets.

Reviewer 2: ```suggestion
* Private - API server endpoint and application routes are private. Private ROSA Classic clusters may use some public subnets, but no control plane or worker nodes are deployed in public subnets.
```

> * Private with AWS PrivateLink - API server endpoint and application routes are private. Public subnets or NAT gateways are not required in your VPC for egress. ROSA SRE management uses AWS PrivateLink.

Reviewer 3: correct, NAT gateways are required.

> * Private with AWS PrivateLink - API server endpoint and application routes are private. Public subnets or NAT gateways are not required in your VPC for egress. ROSA SRE management uses AWS PrivateLink.

Reviewer: Is this inaccurate?
https://docs.openshift.com/rosa/rosa_install_access_delete_clusters/rosa-aws-privatelink-creating-cluster.html#osd-aws-privatelink-about_rosa-aws-privatelink-creating-cluster

> * Private with AWS PrivateLink - API server endpoint and application routes are private. Public subnets or NAT gateways are not required in your VPC for egress. ROSA SRE management uses AWS PrivateLink.

Reviewer 2: > correct, NAT gateways are required.

Wait this is incorrect. No, public subnets are not required, NAT gateways are not required either. We just have a list of required egresses that a customer could be satisfying with a Transit Gateway or VPC peering instead for example.

> * Private with AWS PrivateLink - API server endpoint and application routes are private. Public subnets or NAT gateways are not required in your VPC for egress. ROSA SRE management uses AWS PrivateLink.

Author: Thank you all for the feedback. I will leave the section as is.

> The following image depicts the architecture of both public and private networks.

Reviewer: The following image depicts the architecture of ROSA Classic cluster deployed with both public and private networks.

.ROSA Classic deployed on public and private networks
image::156_OpenShift_ROSA_Arch_0621_private_public_classic.png[ROSA deployed on public and private networks]

ROSA Classic clusters include infrastructure nodes where OpenShift components such as the ingress controller, image registry, and monitoring are deployed. The infrastructure nodes and the OpenShift components deployed on them are managed by ROSA Service SREs.

The following types of clusters are available with ROSA Classic:

* Single zone cluster - The control plane and worker nodes are hosted on a single availability zone.

> * Multi-zone cluster - The control plane is hosted on three availability zones with an option to run worker nodes on one or three availability zones.

Reviewer 4: This is not accurate. For multi-zone cluster,if set it to multi zone at the cluster creation then both control plane and default worker pool will be multi zone. Unless customer create additional machinepool with `--subnet` or `--availability-zone` set.

Author: Again, @bchandra-ocp - Can you please confirm?

Reviewer: Is this inaccurate? 
https://docs.openshift.com/rosa/rosa_install_access_delete_clusters/rosa-aws-privatelink-creating-cluster.html#osd-aws-privatelink-about_rosa-aws-privatelink-creating-cluster

Reviewer: The statement I provided is correct. 

Here I'm describing the architecture not how cluster is created by default. ROSA clusters can have machine pool in single AZ.  How the cluster is created by default (i.e., 3 AZs for control plane and 3 AZs for machine pools) and how machine pools can be created (3 AZs or 1 AZ) should be explained in the respective docs pages.

Author: Thank you, I will leave it as is.

// Module included in the following assemblies:
//
// * rosa_architecture/rosa_architecture_sub/rosa-architecture-models.adoc

[id="rosa-classic-architecture_{context}"]
= ROSA Classic architecture

In ROSA Classic, both the control plane and the worker nodes are deployed in the customer's VPC subnets.

[id="rosa-classic-architecture-networks_{context}"]
== ROSA Classic architecture on public and private networks

With ROSA Classic, you can create your clusters that are accessible over public or private networks.

Customers can customize access patterns for their API server endpoint and Red Hat SRE management in the following ways:

* Public - API server endpoint and application routes are internet-facing.

* Private - API server endpoint and application routes are private. Private ROSA Classic clusters use some public subnets, but no control plane or worker nodes are deployed in public subnets.

> * Private with AWS PrivateLink - API server endpoint and application routes are private. Public subnets or NAT gateways are not required in your VPC for egress. ROSA SRE management uses AWS PrivateLink.

Author: I will make a note to do this as a future update across our product docs.
